src/utils.ts

```diff
@@ -0,0 +1,50 @@
+import * as childProcess from 'child_process';
+import * as fs from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+
+export async function withTempDirectory<T>(fn: (directory: string) => Promise<T>): Promise<T> {
+  const directory = await fs.mkdtemp(path.resolve(os.tmpdir(), 'electron-download-'));
+
+  let result: T;
+  try {
+    result = await fn(directory);
+  } catch (err) {
+    await fs.remove(directory);
+    throw err;
+  }
+
+  await fs.remove(directory);
+  return result;
+}
```

Comment:
This vs. a dependency like `tmp-promise` 🤔

Reply:
The less deps the better IMO